### ISSUE TITLE
Code Hangs during the implementation of DDP 

### ISSUE THREAD

**Issue by Mukil07** [optuna/optuna#5917]
### Don't use GitHub Issues to ask support questions.

The code works perfectly fine for single gpu and also with multigpu (DDP). But while using optuna, the code hangs until the GPU processes gets killed at this line `model = DDP(model)`. 

I have followed the same procedure followed by this example 
https://github.com/optuna/optuna-examples/blob/main/pytorch/pytorch_distributed_simple.py

Im trying to tune only the learning rate and not the model. Also using custom dataloader, and train loops.
